File: kotlinjs/old/Resource.kt


abstract class Resource() : RealTile() {
    override var occ = true
    override var use = false
    override var drive = false
    override var traverse = false
    override var owner = 0
    override var minCost = 0
    override var gasCost = 0
    override var nrgCost = 0
}

File: typescript/src/PlanetTS1/Model/Abstract/Resource.ts


abstract class Resource extends RealTile {
    occ : boolean
    use : boolean
    drive : boolean
    traverse : boolean
    owner : number
    minCost : number
    gasCost : number
    nrgCost : number
    abort : boolean
    commands : Array<string>

    constructor() {
        super()
        this.occ = true
        this.use = false
        this.drive = false
        this.traverse = false
        this.owner = 0
        this.minCost = 0
        this.gasCost = 0
        this.nrgCost = 0
        this.abort = false
        this.commands = new Array<string>("")
    }
}